rnmodules/kb-common/src/MediaUtils.swift

```diff
@@ -3,185 +3,309 @@ import AVFoundation
-                completion(nil)
-            } catch {
-                completion(error)
+            exportSession.exportAsynchronously {
+                timer.invalidate()
+                exportError = exportSession.error
+                semaphore.signal()
+            }
+        } else {
+            exportSession.exportAsynchronously {
+                exportError = exportSession.error
+                semaphore.signal()
             }
         }
+        
+        semaphore.wait()
+        
+        if let error = exportError {
+            throw MediaUtilsError.videoProcessingFailed("Export failed: \(error.localizedDescription)")
+        }
+        
+        guard exportSession.status == .completed else {
+            throw MediaUtilsError.videoProcessingFailed("Export session failed with status: \(exportSession.status)")
+        }
     }
     
-    @objc static func processVideo(fromOriginal url: URL, completion: @escaping ProcessMediaCompletion) {
-        let basename = url.deletingPathExtension().lastPathComponent
-        let parent = url.deletingLastPathComponent()
-        let normalVideoURL = parent.appendingPathComponent("\(basename).scaled.mp4")
+    private static func getFileSize(for url: URL) -> Int64 {
+        do {
+            let attributes = try FileManager.default.attributesOfItem(atPath: url.path)
+            return attributes[.size] as? Int64 ?? 0
+        } catch {
+            return 0
+        }
+    }
+    
+    private static func scaleDownCGImageSource(_ img: CGImageSource, dstURL: URL, options: CFDictionary) throws {
+        guard let scaledRef = CGImageSourceCreateThumbnailAtIndex(img, 0, options) else {
+            throw MediaUtilsError.imageProcessingFailed("Failed to create thumbnail")
+        }
         
-        let asset = AVURLAsset(url: url)
-        let generateImg = AVAssetImageGenerator(asset: asset)
-        generateImg.appliesPreferredTrackTransform = true
+        guard let scaled = UIImage(cgImage: scaledRef).jpegData(compressionQuality: MediaProcessingConfig.imageCompressionQuality) else {
+            throw MediaUtilsError.imageProcessingFailed("Failed to create JPEG data")
+        }
         
         do {
-            _ = try generateImg.copyCGImage(at: CMTimeMake(value: 1, timescale: 1), actualTime: nil)
+            try scaled.write(to: dstURL)
         } catch {
-            completion(error, nil)
-            return
+            throw MediaUtilsError.fileOperationFailed("Failed to write scaled image: \(error.localizedDescription)")
+        }
+    }
+    
+    private static func stripImageExif(at url: URL) throws {
+        guard let cgSource = CGImageSourceCreateWithURL(url as CFURL, nil) else {
+            throw MediaUtilsError.imageProcessingFailed("Failed to create image source")
         }
         
-        if !needScaleDownAsset(asset) {
-            stripVideoExifAndComplete(asset: asset, originalURL: url) { error in
```

Comment:
this whole flow in the old code is strange, we scale down, then transcode so we handle it twice sometimes